libusb/core.c

```diff
@@ -2362,6 +2361,7 @@ int API_EXPORTEDV libusb_set_option(libusb_context *ctx,
 	default:
 		return LIBUSB_ERROR_INVALID_PARAM;
 	}
+	va_end(ap);
```

Comment:
there is multiple `return` statements before `va_end` and after `va_start`, and I believe the C standard states, that if the function that invoked `va_start` doesn't invoke `va_end` before return - the behavior is undefined.

Reply:
effectively it makes this fix a lot more complicated

Reply:
Thanks for the review. That is a good point. The man page doesn't warn about it, but I can imagine possible memory leaks.

Reply:
I have attempted a do loop to make single point of return, the diff is best viewed with git diff -w (without whitespace since I had to indent everything). The alternative would be a lot of "else if" and it would be harder to get it right and difficult to read.

Reply:
that should work